src/test-gui/java/de/qabel/desktop/ui/sync/SyncStepdefs.java

```diff
@@ -0,0 +1,167 @@
+    protected FXMLView getView() {
+        return new SyncView();
+    }
+
+    @Override
+    @After("@sync")
+    public void tearDown() throws Exception {
+        executor.shutdownNow();
+        try {
+            if (Files.isDirectory(tmpDir)) {
+                FileUtils.deleteDirectory(tmpDir.toFile());
+            }
+        } catch (IOException e) {
+            if (Files.isDirectory(tmpDir)) {
+                FileUtils.deleteDirectory(tmpDir.toFile());
+            }
+        }
+        super.tearDown();
+    }
+
+    private Path toTmpPath(String folder) throws Exception {
+        return tmpDir.resolve(folder);
+    }
+
+    @Given("^a sync (.+) from '(.+)' to '(/?.+)'$")
+    public void aSyncAFromLocalFolderToRemoteFolder(String name, String from, String to) throws Throwable {
+        localPath = toTmpPath(from);
+        remotePath = BoxFileSystem.get(to);
+        syncName = name;
+        BoxSyncConfig syncConfig = new DefaultBoxSyncConfig(
+            syncName,
+            localPath,
+            remotePath,
+            identity,
+            account
+        );
+        volume = boxVolumeFactory.getVolume(account, identity);
+        boxSyncRepository.save(syncConfig);
+        waitUntil(() -> syncConfig.getSyncer() != null && syncConfig.getSyncer().isSynced(), 30000L);
+    }
+
+    @And("^a synced file '(.+)'$")
+    public void aSyncedFileTestfile(String filename) throws Throwable {
+        Files.write(localPath.resolve(filename), ".".getBytes());
+        waitUntil(() -> {
+            BoxNavigation navigation = volume.navigate();
+            for (int i = 0; i < remotePath.getNameCount(); i++) {
+                String folder = remotePath.getName(i).getFileName().toString();
+                navigation = navigation.navigate(folder);
+            }
+            return navigation.hasFile(filename);
+        }, 10000L);
+    }
+
+    @When("^I change the local folder to '(.+)'$")
+    public void iChangeTheLocalFolder(String newPath) throws Throwable {
+        SyncItemPage itemPage = page.getSync(syncName);
+        itemPage.edit().enterLocalPath(toTmpPath(newPath).toString()).save();
+    }
+
+    @Then("^'(.+)' exists in local '(.+)'$")
+    public void fileExistsInLocal(String filename, String localPath) throws Throwable {
+        Path targetFile = toTmpPath(localPath).resolve(filename);
+        waitUntil(() -> Files.exists(targetFile), 10000L, () -> "file not found: " + targetFile);
+    }
+
+    @And("^'(.+)' exists in remote '(.+)'$")
+    public void fileExistsInRemote(String filename, String remotePath) throws Throwable {
+        Path remote = BoxFileSystem.get("/", remotePath);
+        final BoxNavigation remoteNavigation = navigate(remote);
+        waitUntil(() -> remoteNavigation.hasFile(filename));
+    }
+
+    private BoxNavigation navigate(Path remote) throws QblStorageException {
+        BoxNavigation navigation = volume.navigate();
+        for (int i = 0; i < remote.getNameCount(); i++) {
```

Comment:
Didn't know it yet but I just found out that Path implements Iterable .. so looping over that path itself will provide its names. Going to fix that...